pillow_heif/_pillow_heif.c

```diff
@@ -738,12 +738,13 @@ PyObject* _CtxDepthImage(struct heif_image_handle* main_handle, heif_item_id dep
                             int remove_stride, int hdr_to_16bit, PyObject* file_bytes) {
     struct heif_image_handle* depth_handle;
     if (check_error(heif_image_handle_get_depth_image_handle(main_handle, depth_image_id, &depth_handle))) {
-        Py_RETURN_NONE;
+        return NULL;
     }
     CtxImageObject *ctx_image = PyObject_New(CtxImageObject, &CtxImage_Type);
     if (!ctx_image) {
         heif_image_handle_release(depth_handle);
-        Py_RETURN_NONE;
+        PyErr_SetString(PyExc_RuntimeError, "could not create CtxImage object");
+        return NULL;
```

Comment:
I think `PyObject_New` can only fail when it fails to allocate to heap. So maybe this should also raise a No-Memory error `PyErr_NoMemory()` ? (In [a post from 23 years ago](https://python-dev.python.narkive.com/nHE4i5zQ/use-of-pyobject-new), I read that `PyObject_New` returning `NULL` should already set a memory exception. So it might be safest to just return `NULL` and nothing else here.)

Reply:
This still seems to be the case: https://github.com/python/cpython/blob/2f8301cbfbdd2976d254a4a772b4879069dd4298/Objects/object.c#L459-L468

This means that we either _have_ to throw an error every time `PyObject_New` fails, or we have to call [`PyErr_Clear`](https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Clear) if we want to ignore that error. I think it's okay to give up when there's a `MemoryError`.

I've updated all instances (I could find here) to simply propagate the exception.

Reply:
> I think `PyObject_New` can only fail when it fails to allocate to heap. So maybe this should also raise a No-Memory error `PyErr_NoMemory()` ? (In [a post from 23 years ago](https://python-dev.python.narkive.com/nHE4i5zQ/use-of-pyobject-new), I read that `PyObject_New` returning `NULL` should already set a memory exception. So it might be safest to just return `NULL` and nothing else here.)

agree, I think we should just return `NULL` and even do not call `PyErr_SetString`

numpy src:
https://github.com/numpy/numpy/blob/a5cb32794df5dce20a3670cf11243e397859bbca/numpy/f2py/src/fortranobject.c#L132
pillow src:    
https://github.com/python-pillow/Pillow/blob/97438cb8fd51f016117953072a7a68c2f74ccc74/src/decode.c#L69